src/io/parsers/MOL2Parser.js

```diff
@@ -0,0 +1,265 @@
+      }
+      const atomId = parseInt(parsedStr[0], 10);
+      const atomName = parsedStr[1];
+
+      const x = parseFloat(parsedStr[2]);
+      const y = parseFloat(parsedStr[3]);
+      const z = parseFloat(parsedStr[4]);
+
+      const element = parsedStr[5].split('.')[0].toUpperCase();
+
+      let charge = 0;
+      if (parsedStr.length >= 9) {
+        charge = parseFloat(parsedStr[8]) | 0;
+      }
+
+      let chain = this._chain;
+      if (!chain) {
+        // .mol2 may contain information about multiple molecules, but they can't be visualized
+        // at the same time now. There is no need to create different chain IDs then.
+        this._chain = chain = this._complex.getChain('A') || this._complex.addChain('A');
+        this._residue = null;
+      }
+      if (!this._setResidue(parsedStr)) {
+        continue;
+      }
+
+      // These fields are not listed in mol2 format. Set them default.
+      // Atoms and het atoms doesn't differ in .mol2,
+      // but het atoms have special residues. It can be used in next updates
+      const het = false;
+      const altLoc = ' ';
+      const occupancy = 1.0;
+      const tempFactor = 0.0;
+      const type = Element.getByName(element);
+      const role = Element.Role[atomName];
+
+      const xyz = new THREE.Vector3(x, y, z);
+      this._residue.addAtom(atomName, type, xyz, role, het, atomId, altLoc, occupancy, tempFactor, charge);
+    }
+  }
+
+  _setResidue(parsedStr) {
+    let resSeq = 1;
+    let resName = 'UNK'; // The same meaning has '<0>' in some mol2 files
+
+    if (parsedStr.length >= 7) {
+      resSeq = parseInt(parsedStr[6], 10);
+    }
+    if (parsedStr.length >= 8 && parsedStr[7] !== '<0>') {
+      resName = parsedStr[7].replace(resNumberRegex, '');
+    }
+    if (this.settings.now.nowater) {
+      if (resName === 'HOH' || resName === 'WAT') {
+        return false;
+      }
+    }
+    const residue = this._residue;
+    const chain = this._chain;
+    if (!residue || residue.getSequence() !== resSeq) {
+      this._residue = chain.addResidue(resName, resSeq, 'A');
+    }
+    return true;
+  }
+
+  /* Bond format description
+   * bondId originAtomId targetAtomId bondType [statusBits]
+   */
+  _parseBonds(stream, bondsNum) {
+    let curStr = stream.getHeaderString('BOND');
+
+    for (let i = 0; i < bondsNum; i++) {
+      curStr = stream.getNextString();
+      const parsedStr = splitToFields(curStr);
+
+      if (parsedStr.length < 3) {
+        throw new Error('MOL2 parsing error: Missing information about bonds!');
+      }
+
+      let originAtomId = parseInt(parsedStr[1], 10);
+      let targetAtomId = parseInt(parsedStr[2], 10);
+      const bondType = parsedStr[3];
+
+      if (originAtomId > targetAtomId) {
+        [originAtomId, targetAtomId] = [targetAtomId, originAtomId];
+      }
+      this._complex.addBond(originAtomId, targetAtomId,
+        (bondType in typeMap) ? orderMap[bondType] : 1,
```

Comment:
Please do not use `in`, ever :) Don't be afraid to access a property and get `undefined` as a result. I would say `orderMap[bondType] || 1` is much better but the map contains zero values. Do we really have zeroes there? Then why not zero here?

Reply:
That's my fault. It is better to compare orderMap[bondType] to zero, the mol2 documentation does not provide any other possible types of bonds and the corresponding orders, so we don't have to try to set the default value to 1 if we meet undefined type of a bond.